contracts/mechs/nevermined/BalanceTrackerNvmSubscriptionNative.sol

```diff
@@ -79,31 +79,25 @@ contract BalanceTrackerNvmSubscriptionNative is BalanceTrackerFixedPriceNative {
         // Get requester actual subscription balance
         uint256 subscriptionBalance = IERC1155(subscriptionNFT).balanceOf(requester, subscriptionTokenId);
 
-        // Adjust requester balance with maxDeliveryRate credits
-        balance += maxDeliveryRate;
+        uint256 totalBalance = balance + subscriptionBalance;
 
         // Check the request delivery rate for a fixed price
-        if (subscriptionBalance < balance) {
-            revert InsufficientBalance(subscriptionBalance, balance);
+        if (totalBalance < maxDeliveryRate) {
+            revert InsufficientBalance(totalBalance, maxDeliveryRate);
         }
 
-        return balance;
-    }
-
-    /// @dev Adjusts final requester balance accounting for possible delivery rate difference (credit).
-    /// @param requester Requester address.
-    /// @param rateDiff Delivery rate difference.
-    /// @return Adjusted balance.
-    function _adjustFinalBalance(address requester, uint256 rateDiff) internal virtual override returns (uint256) {
-        uint256 balance = mapRequesterBalances[requester];
-
-        // This must never happen as max delivery rate is always bigger or equal to the actual delivery rate
-        if (rateDiff > balance) {
-            revert Overflow(rateDiff, balance);
+        // Calculate how many actual credits need to burn accounting for discount balance
+        if (balance > maxDeliveryRate) {
+            balance -= maxDeliveryRate;
+        } else {
+            maxDeliveryRate -= balance;
+            balance = 0;
+            IERC1155(subscriptionNFT).burn(requester, subscriptionTokenId, maxDeliveryRate);
         }
```

Comment:
self note: describing the change

1. `balance` without modification is the `mapMechBalances[requester]` (no initial native token input amount) which now only represents the accumulated discounted rates in the request/delivery flow.

2. when creating a request, the credits are taken from `mapMechBalances[requester]` and if there is any surplus, the rest of the credit is taken from the `requester`'s credit balance of `subscriptionTokenId` in `subscriptionNFT` by burning that surplus amount.

3. `totalBalance` is basically the sum of returned discounted credits and actual credits in the `subscriptionNFT`